demosymfonyform/demosymfonyform.php

```diff
@@ -0,0 +1,60 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+declare(strict_types=1);
+
+use PrestaShop\PrestaShop\Adapter\SymfonyContainer;
+
+class DemoSymfonyForm extends Module
+{
+    public function __construct()
+    {
+        $this->name = 'demosymfonyform';
+        $this->author = 'PrestaShop';
+        $this->version = '1.0.0';
+        $this->need_instance = 0;
+
+        $this->bootstrap = true;
+        parent::__construct();
+
+        $this->displayName = $this->trans('Demo symfony form configuration', array(), 'Modules.DemoSymfonyForm.Admin');
+        $this->description = $this->trans(
+            'Module created for the purpose of showing existing form types within PrestaShop',
+            array(),
+            'Modules.DemoSymfonyForm.Admin'
+        );
+
+        $this->ps_versions_compliancy = array('min' => '1.7.8.0', 'max' => _PS_VERSION_);
+    }
+
+
+    public function getContent()
+    {
+        $route = SymfonyContainer::getInstance()->get('prestashop.router')->generate('demo_configuration_form');
```

Comment:
Hoho nice trick I did not think about that 😄 

I'm wondering if it would be possible to go further (just a crazy idea, not to be done in this PR)
```
public function getContent()
    {
        $response = SymfonyContainer::getInstance()->dispatcher->dispatch($myRequest);
        return $response->getContent();
    }
```

Reply:
It might work but you would need to build the request object manually (which would probably lack parameters set via listeners, like route attributes and so on), so I'm not sure everything would work as expected (ex: resolving parameters automatically for Grid controllers)

Besides you'd still get a legacy url like 'AdminModuleController?action=configure&module=demosymfonyform' (probably not accurate but that's the idea)

The advantage of using the router is that you can redirect to a true route with a proper controller url

Reply:
That said, I think it would be better to `router` instead of `prestashop.router`, we indeed have a custom router in PrestaShop core but the `router` is aliased to it So it would keep the module more generic in case we one day get rid of our custom one

Reply:
Didn't know about the router, will implement that!

 As regarding request, I agree that it shouldn't be done in this PR. I think          $route = SymfonyContainer::getInstance()->get('prestashop.router')->generate('demo_configuration_form');  is simpliest way to implement routing to your controller and that's what most module developers would be looking for.